content/en/docs/configuring-sdk/endpoints.md

```diff
@@ -0,0 +1,105 @@
+client is invoking your endpoint resolver.
+
+An endpoint resolver can use the [EndpointNotFoundError]({{< apiref "aws#EndpointNotFoundError" >}}) sentinel error
+value to trigger fallback resolution to the service clients default resolution logic. This allows you to selectively
+override one or more endpoints seamlessly without having to handle fallback logic.
+
+If your endpoint resolver implementation returns an error other than `EndpointNotFoundError`, endpoint resolution will
+stop and the service operation will return an error to your application.
+
+## Examples
+
+### Overriding Endpoint with Fallback
+
+The following code snippet shows how a single service endpoint can be overridden for {{% alias service=DDB %}} with
+fallback behavior for other endpoints:
+
+```go
+customResolver := aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+    if service == dynamodb.ServiceID && region == "us-west-2" {
+        return aws.Endpoint{
+            PartitionID:   "aws",
+            URL:           "https://test.us-west-2.amazonaws.com",
+            SigningRegion: "us-west-2",
+        }, nil
+    }
+    // returning EndpointNotFoundError will allow the service to fallback to it's default resolution
+    return aws.Endpoint{}, &aws.EndpointNotFoundError{}
+})
+
+cfg, err := config.LoadDefaultConfig(config.WithEndpointResolver(customResolver))
+```
+
+### Overriding Endpoint Without Fallback
+
+The following code snippet shows how a single service endpoint can be overridden for {{% alias service=DDB %}} without
+fallback behavior for other endpoints:
+
+```go
+customResolver := aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+    if service == dynamodb.ServiceID && region == "us-west-2" {
+        return aws.Endpoint{
+            PartitionID:   "aws",
+            URL:           "https://test.us-west-2.amazonaws.com",
+            SigningRegion: "us-west-2",
+        }, nil
+    }
+    return aws.Endpoint{}, fmt.Errorf("unknown endpoint requested")
+})
+
+cfg, err := config.LoadDefaultConfig(config.WithEndpointResolver(customResolver))
+```
+
+### Immutable Endpoints
+
+{{% pageinfo color="warning" %}}
+Setting an endpoint as immutable may prevent some service client features from functioning correctly, and could result
+in undefined behavior. Caution should be taken when defining an endpoint as immutable.
+{{% /pageinfo %}}
+
+Some service clients, such as {{% alias service=S3 %}}, can modify the endpoint returned by the resolver for certain
+service operations. For example, the {{% alias service=S3 %}} will automatically handle
+[Virtual Bucket Addressing](https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html) by mutating the
+resolved endpoint. You can prevent the SDK from mutating your custom endpoints by setting `HostnameImmutable` to
+`false`. For example:
+
+```go
```

Comment:
maybe link to the HostImmutable api reference docs?  `apiref aws#Endpoint.HostnameImmutable`